### PR TITLE
Pong in Vanilla

### PR DESCRIPTION
**Pull request by hankc97**

### CODE REVIEW

File: pongObjects_Methods.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+// get canvas properties 
+const cvs = document.getElementById("PongTable");
```
Comment (EChu16) [inline pongObjects_Methods.js]: this is vanilla JS so you should use var for backwards compatibility on older browsers

File: pongObjects_Methods.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+};
+//Draw TextScore
+function drawTextscore(x, y, color, text){
+    ctx.fillStyle = color;
+    ctx.font = "45px fantasy"; 
+    ctx.fillText(text, x, y);
+}
+//Draw the net using rectangles incremented every 15 pixels
+function drawNet(){
+    for (let i = 0; i <= cvs.height; i+=15){ // Instead of drawing each rectangle we can create a loop to draw at every 15th pixel for the total length of the canvas height
```
Comment (EChu16) [inline pongObjects_Methods.js]: why not draw a vertical line (one long rectangle in height) instead

File: pongObjects_Methods.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    return ball.right > paddle.left && ball.bottom > paddle.top && ball.left < paddle.right && ball.top < paddle.bottom; //return true is all conditions are met
+}
+function reset(){
+    ball.x = cvs.width/2;
+    ball.y = cvs.height/2;
+
+    ball.speed = 5;
+// we only need to change the velocityX direction, i.e if user loses it means initially the ball was moving left 
+// so we start the ball moving right, vice versa
+    ball.velocityX = -ball.velocityX
```
Comment (EChu16) [inline pongObjects_Methods.js]: nit: ball.velocityX *= -1

File: pongObjects_Methods.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+// so we start the ball moving right, vice versa
+    ball.velocityX = -ball.velocityX 
+}
+function update(){
+    ball.x += ball.velocityX; //Increments the ball x,y direction by 5 or by velocityX/Y; This will send the ball towards the bottom right 
+    ball.y += ball.velocityY; //Think of bottom right as our +X, +Y 
+
+    //Without the aiDifficulty var, the aiPaddle will always be in the center of the ball, the aiPaddle will trail a little behind
+    //This can give the user a chance to score!
+    let aiDifficulty = 0.1;
```
Comment (EChu16) [inline pongObjects_Methods.js]: nit: would wrap AI logic into a separate function. same with ball movement update

File: pongObjects_Methods.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
+// get canvas properties 
+const cvs = document.getElementById("PongTable");
+const ctx = cvs.getContext("2d");
+
+// Create User Paddle
+// x, y coordinates are used to start where the paddle will be drawn
+//width, height is the size of paddle itself
+const userPad = {
```
Comment (EChu16) [inline pongObjects_Methods.js]: similar variables, similar code. cleaner way is to create a function that returns this object.

i.e 
`function createPaddle(x, y) {
    return {
        x: x,
        y: y,
        width: 10,
        height: 100,
        color: 'WHITE',
        score: 0,
    }
}`

and then do

`var userPaddle = createPaddle(0, cvs.height/2 - 100/2)`
`var aiPaddle = createPaddle(cvs.width - userPad.width, cvs.height/2 - 100/2)`

File: pongObjects_Methods.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    y : cvs.height / 2,
+    speed : 5,
+    velocityX : 5, 
+    velocityY: 5,
+    radius : 10,
+    color : "WHITE"
+}
+//Draw Rectangle takes in all 5 properties above
+//x,y starting point of draw; w,h actual size of rect; 
+function drawRectangle(x,y,w,h,color){
```
Comment (EChu16) [inline pongObjects_Methods.js]: nit: better name is drawPaddle

File: pongObjects_Methods.js (hunk 1 above)
Comment (hankc97) [reply pongObjects_Methods.js]: var pongTable = $('#PongTable')

**Review by EChu16 — APPROVED** [review]

**Pull request merged**